lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.cpp

```diff
@@ -2937,4 +2940,87 @@ llvm::Expected<lldb::addr_t> GetTaskAddrFromThreadLocalStorage(Thread &thread) {
   return task_addr;
 #endif
 }
+
+namespace {
+
+struct TaskStatusRecord {
+  Process *process;
+  addr_t addr;
+
+  operator bool() const { return addr && addr != LLDB_INVALID_ADDRESS; }
+
+  static constexpr offset_t FlagsOffset = 0x0;
+  static constexpr offset_t ParentOffset = 0x8;
+  static constexpr offset_t TaskNameOffset = 0x10;
+
+  enum Kind : uint64_t {
+    TaskName = 6,
+  };
+
+  uint64_t getKind(Status &status) {
+    if (status.Success())
+      return process->ReadUnsignedIntegerFromMemory(
+          addr + FlagsOffset, sizeof(uint64_t), UINT64_MAX, status);
+    return UINT64_MAX;
+  }
+
+  std::optional<std::string> getName(Status &status) {
+    if (getKind(status) != Kind::TaskName)
+      return {};
+
+    addr_t name_addr =
+        process->ReadPointerFromMemory(addr + TaskNameOffset, status);
+    if (!status.Success())
+      return {};
+
+    std::string name;
+    process->ReadCStringFromMemory(name_addr, name, status);
+    if (status.Success())
+      return name;
+
+    return {};
+  }
+
+  TaskStatusRecord getParent(Status &status) {
```

Comment:
For future-proofing you could convert these functions to Expected like so:
```
llvm::Expected<TaskStatusRecord> getParent() {
     addr_t parent = LLDB_INVALID_ADDRESS;
     Status status;
     if (*this)
       parent = process->ReadPointerFromMemory(addr + ParentOffset, status);
     if (Status.Fail())
       return Status.takeError();
     return {process, parent};
   }

Reply:
Because these functions perform a number of `Read*` calls to process, I kept `Status` as the means of communicating errors. It makes the code more concise while still having the error handling/control. After the set of Reads (a logical transaction of sorts) have been completed (or failure occurred), the `Status` is then converted to `Expected`. This pattern I used elsewhere in support of inspection of other concurrency types.

Reply:
I'll leave a comment describing this.

Reply:
I must admit that this pattern also gave me a hard time convincing me everything was correct. Especially reading something like `if (getKind(status) != Kind::TaskName)`, it reads like we're extracting the kind from the status